internal/pkg/agent/application/dispatcher/dispatcher_test.go

```diff
@@ -248,10 +251,11 @@ func TestActionDispatcher(t *testing.T) {
 		select {
 		case err := <-d.Errors():
 			t.Fatalf("Unexpected error: %v", err)
-		case <-time.After(200 * time.Microsecond):
-			// we're not expecting any reset,
+		case <-calledCh:
+			// Handle was called, expected
+		case <-time.After(1 * time.Second):
```

Comment:
You could probably make this even shorter, given that the `assert.Eventually` before had a timeout of `100*time.Millisecond`?

Reply:
I just like to keep it longer because it returns as soon as it happens. This is just to be sure that in the case that runner is slow that we don't hit that timeout just because of that.

Reply:
You okay with me leaving it as is?

Reply:
Yes, OK to leave as-is.